### ISSUE TITLE
Romerol (Zombie Reagent) Doesn't Work

### ISSUE THREAD

**Issue by Kyep** [ParadiseSS13/Paradise#12392]
**Issue Description**:
Romerol reagent https://github.com/ParadiseSS13/Paradise/pull/11535 does not work at all.
It never revives anyone as a zombie.

**What did you expect to happen**: 
Someone who (A) is injected with romerol and (B) dies, will rise up as a zombie after a few minutes.

**What happened instead**:
They die, and after awhile they are healed, but they do not become a zombie.
This leaves them in a weird state where the are dead with no damage, but are not a zombie.

**Why is this bad/What are the consequences**:
There's no point in having Romerol in the game if it doesn't work.

**Steps to reproduce the problem**:
1) Spawn a human
2) Possess the human so it has a mind
3) Add 5u of Romerol to the mob
4) Call 'death' proc on the mob (or do enough damage to the mob that it dies, but this generates a second bug, see below)
5) Wait 2 minutes
6) Observe that the mob does not re-animate

**How Romerol is MEANT to work**:
1) Reagent defined here:
https://github.com/ParadiseSS13/Paradise/blob/7f31e4f7a364ad484c2c00b38a3a319326d6624f/code/modules/reagents/chemistry/reagents/toxins.dm#L240
2) Adds /obj/item/organ/internal/zombie_infection/nodamage organ defined here: https://github.com/ParadiseSS13/Paradise/blob/070371134a0ef99fda7fe37ea8ab83c25ccff34a/code/modules/zombie/organs.dm#L1
3) New() adds it to processing, processing keeps checking if it should start a timer (victim is dead) when it does uses addtimer to add a callback to call zombify() https://github.com/ParadiseSS13/Paradise/blob/070371134a0ef99fda7fe37ea8ab83c25ccff34a/code/modules/zombie/organs.dm#L66
4) zombify() converts the victim into an actual zombie

**How Romerol actually FAILS to work**:
1) Romerol doesn't work on mobs with no mind, due to a hidden mind check in grab_ghost. It will still heal the damage on these mobs, but it won't reanimate them, which leaves them in a weird state where they're dead despite having no damage.
https://github.com/ParadiseSS13/Paradise/blob/f4fd1aa9615db084eca47a73c9a54f6ab6748adc/code/modules/mob/mob.dm#L1151
2) Romerol also fails on any mob with a token amount of damage, because zombify() calls update_revive() https://github.com/ParadiseSS13/Paradise/blob/acd191a14d5a85fdbbecc809debb6a107e1d8708/code/modules/mob/living/stat_states.dm#L42 which checks can_be_revived https://github.com/ParadiseSS13/Paradise/blob/acd191a14d5a85fdbbecc809debb6a107e1d8708/code/modules/mob/living/stat_states.dm#L33 which returns FALSE if (health <= HEALTH_THRESHOLD_DEAD) which means (health < -100) IE the vast majority of the time a newcrit species is dead all of this returns FALSE and fails.
3) Aside from not working on mindless mobs (meaning you can't test it on a monkey, or spawned in human that's never been player controlled) and not working on mobs with more than a token amount of damage, there may be other issues with it beyond this, but I stopped detailed testing at this point due to the abundance of issues.

In short: Romerol is broken on multiple levels.

**Comment by Couls** [comment]
romerol zombies handle_death proc removes the infection
on TG its not an issue because zombies can't actually die due to a trait
though here handle_death is called on death and zombies die here and revive

also, in the zombify proc all the damaged is healed before calling update_revive
as shown here:
`	owner.setToxLoss(0)
	owner.setOxyLoss(0)
	owner.setBrainLoss(0)
	owner.setCloneLoss(0)
	owner.SetLoseBreath(0)
	owner.heal_overall_damage(INFINITY, INFINITY, TRUE, TRUE, FALSE)
	owner.setStaminaLoss(0)`
so by the time update_revive is called, the zombie should not have any damage and be below HEALTH_THRESHOLD_DEAD

zombies should have a mind as long as they were played by a player, I used the same logic SR does for grabbing the ghost and updating revive, romerol was not really meant for monkeys or humans that aren't player controlled. It was probably designed as such by TG to prevent exploiting I assume.

**Comment by Kyep** [comment]
If mindless mobs are meant to be immune, then that should be obvious when you try to inject them with it. It shouldn't be something you find out after its reached the point of trying to zombify them, getting halfway though, and then silently failing and aborting.

**Comment by Couls** [comment]
to explain why kasparovy's rework would fix an issue with romerol, the organ upon insertion saves the old race of the person with the organ. Then the zombify proc converts them into a zombie, our code for changing species replaces every organ with the organ of the species. When it replaces the infection organ it overwrites the old species which is now a zombie. kasparovy's rework changes it so changing species doesn't replace every organ like it does currently and that will fix the issue of romerol zombies being unable to be deconverted.

**Comment by Couls** [comment]
A check can be added for mindless mobs

**Comment by farie82** [comment]
Can be closed
